code/modules/mob/living/carbon/human/human_life.dm

```diff
@@ -47,6 +47,9 @@
 	if(player_logged > 0 && stat != DEAD && job)
 		handle_ssd()
 
+	if(mind && !mind.antag_datums) // Only humans with a mind and no antags
+		handle_light_adjustment()
+
```

Comment:
It's now called every `Life` of humans. I know this isn't the most performant execution, but after quite a lot of research into how darkvision goes, it's either this or calling `update_sight` every life proc instead.
I'll have to profile this later though, before it goes out of draft.

Reply:
does this mean a traitor has eyes that work worse than a normal crewmember?
why the antag check?

Reply:
No it'd mean that antags would have no changes from the current system. Genuinely forgot why I added this, I think it stemmed from a discussion on how this would impact antagonists, and that they probably shouldn't be affected.
If you think it should just be for everyone though, I'd be fine with removing it